pkg/rebuild/maven/maveninfer.go

```diff
@@ -143,6 +111,23 @@ func MavenInfer(ctx context.Context, t rebuild.Target, mux rebuild.RegistryMux,
 	}, nil
 }
 
+// validateCommit is a helper that checks if a given commit contains a valid pom.xml for the package.
+// It returns the directory containing the pom.xml and a log message.
+// If no valid pom.xml is found, it returns an empty directory and a log message indicating the failure reason.
+func validateCommit(commit *object.Commit, t rebuild.Target) (dir string, logMessage string) {
```

Comment:
okay i can't think of a better solution for this pattern so let's rename the function to "findBuildDir" and the logMessage output to "summary" for now.